src/main/java/seedu/bookmark/model/ModelManager.java

```diff
@@ -36,6 +39,7 @@ public ModelManager(ReadOnlyLibrary library, ReadOnlyUserPrefs userPrefs) {
 
         this.library = new Library(library);
         this.userPrefs = new UserPrefs(userPrefs);
+        this.comparator = comparatorGenerator(sortingPrefixGenerator(userPrefs.getSortingPreference()));
```

Comment:
Relating to my comment on using a Prefix instead of String in UserPrefs below, a Prefix would remove the need for `sortingPrefixGenerator()`. This line seems to have too much chaining.